### ISSUE TITLE
Panic in `Uri::try_from`

### ISSUE THREAD

**Issue by Stranger6667** [yescallop/fluent-uri-rs#28]
It should not panic on empty input:

```rust
use fluent_uri::{Uri, UriRef};

let a = UriRef::parse("").expect("Valid URI ref");
let b = Uri::try_from(a);

// thread '<unnamed>' panicked at /.../fluent-uri-0.3.0/src/ri.rs:842:18:
// called `Option::unwrap()` on a `None` value
```

**Comment by yescallop** [comment]
Fixed in 92ba6fdfadf920f0f60145b77b18e23bd409bcf5. Thank you for reporting! I'll publish a new release after I check other parts of the code.

**Comment by Stranger6667** [comment]
Amazing! Thank you for your swift response and fix! :)
